### PR TITLE
Fix MCV compilation issues

### PR DESCRIPTION
**Pull request by paladine**
Old code was writing language twice at the front. Also properly emulates the `\r` that gets modified during compile time.

### CODE REVIEW

File: MBBSEmu/Module/MsgFile.cs [hunk 1]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
+        private enum MsgParseState {
+            NEWLINE,
+            IDENTIFIER,
+            SPACE,
+            BRACKET,
+            JUNK
+        };
+
+        private static bool IsIdentifier(char c) =>
+            Char.IsDigit(c) || (c >= 'A' && c <= 'Z');
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:54]: `char`

File: MBBSEmu/Module/MsgFile.cs [hunk 2]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
+
+        private static bool IsAlnum(char c) =>
+            Char.IsLetterOrDigit(c);
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:57]: `char`

File: MBBSEmu/Module/MsgFile.cs [hunk 3]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
+
+        private static string FixLineEndings(string rawMessage) {
+            var ret = new StringBuilder(rawMessage.Length + 16);
+
+            for (var i = 0; i < rawMessage.Length; ++i)
+            {
+                var c = rawMessage[i];
+                if (c == '\n')
+                {
+                    char next = i < (rawMessage.Length - 1) ? rawMessage[i + 1] : (char)0;
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:67]: var

File: MBBSEmu/Module/MsgFile.cs [hunk 4]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
+            /*
+            * 16-byte final header is
+            *
+            * uint32_t ;   // ptr to languages
+            * uint32_t ;   // ptr to length array, always last
+            * uint32_t ;   // ptr to location list
+            * uint16_t ;   // count of languages
+            * uint16_t ;   // count of messages
+            */
+            using (var writer = File.Open(Path.Combine(_modulePath, FileNameAtRuntime), FileMode.Create))
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:160]: `using var` to simplify the nesting

File: MBBSEmu/Module/MsgFile.cs [hunk 5]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
 
-                    bInVariable = false;
+                    writer.Write(GetStringBytes(message));
+                    offset += message.Length + 1;
+                }
 
-                    //Always null terminate
-                    msCurrentValue.WriteByte(0x0);
+                var numberOfLanguages = 1;
+                var numberOfMessages = messages.Count;
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:181]: not used?

File: MBBSEmu/Module/MsgFile.cs [hunk 6]
```diff
@@ -42,145 +42,179 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
-                    if (MsgValues.ContainsKey(variableName))
-                        variableName += $"_{MsgValues.Count(x=> x.Key.StartsWith(variableName))}";
+                // write out the file lengths if requested
+                if (writeStringLengthTable)
+                {
+                    stringLengthArrayPointer = offset;
 
-                    //Write to Variable Dictionary
-                    MsgValues.Add(variableName, msCurrentValue.ToArray());
+                    foreach (string message in messages)
```
Comment (enusbaum) [inline MBBSEmu/Module/MsgFile.cs:196]: `var message`

**Review by enusbaum — CHANGES_REQUESTED** [review]
With this refactor we can also clean up the imports as well. Don't need Linq + x509

File: MBBSEmu/Module/MsgFile.cs (hunk 1 above)
Comment (paladine) [reply MBBSEmu/Module/MsgFile.cs:54]: isn't this the same thing being a static method?

Comment (enusbaum) [reply MBBSEmu/Module/MsgFile.cs:54]: it's more just a style thing to reference the actual type `char` vs. the alias `Char`

**Review by enusbaum — APPROVED** [review]

**Pull request merged**